### ISSUE TITLE
Socket is leaked and displays ResourceWarning in Python 3

### ISSUE THREAD

**Issue by vsajip** [dropbox/dropbox-sdk-python#78]
With version 7.1.1 on Ubuntu 16.04.1 (64-bit), the following code:
```
def save_to_dropbox(fn, overwrite=False):
    if not DROPBOX_ACCESS_TOKEN:
        return
    dfn = '/dropbox-data/%s' % os.path.basename(fn)
    dbx = dropbox.Dropbox(DROPBOX_ACCESS_TOKEN)
    try:
        data = dbx.files_get_metadata(dfn)
        exists = True
    except dropbox.exceptions.ApiError:
        exists = False
    if not exists or overwrite:
        with open(fn, 'rb') as f:
            data = f.read()
        mode = dropbox.files.WriteMode('overwrite', None)
        dbx.files_upload(data, dfn, mode=mode, mute=True)
    del dbx
```
when run on Python 3, leaks a socket and leads to a `ResourceWarning`:
```
ResourceWarning: unclosed <ssl.SSLSocket fd=4, family=AddressFamily.AF_INET, type=2049, proto=6, laddr=('192.168.0.46', 58782), raddr=('108.160.172.237', 443)>
```
The `108.160.172.237` is a Dropbox server address. The `ResourceWarning` shows up at the `del dbx` line, which I added specifically when testing (without it, the `ResourceWarning` still occurs, but against different lines in the source).

There's no obvious API to tidy up after finishing with the `dbx` instance.

**Comment by greg-db** [comment]
Thanks! We'll look into it.

**Comment by quinox** [comment]
You could make your own session object, call `dropbox.Dropbox(..., session=mysession)` and call `mysession.close()` when you're done. Alternatively you can call `dbx._session.close()` before `del`, but obviously you don't want to be calling protected methods of other people's code.

A `dbx.close()` or `dbx.cleanup()` which does `self._session.close()` would be nice.

**Comment by greg-db** [comment]
Thanks for the additional information and feedback!

**Comment by guilhermecgs** [comment]
any updates on this?

**Comment by greg-db** [comment]
@guilhermecgs No, this is still open for engineering to investigate, but unfortunately I don't have an update on this.

**Comment by greg-db** [comment]
This should now be fixed as of v10.3.0 of the Dropbox Python SDK.

You can find an example of using the Dropbox client with a context manager to clean up network resources here: https://github.com/dropbox/dropbox-sdk-python/blob/master/example/back-up-and-restore/backup-and-restore-example.py#L79

**Comment by vsajip** [comment]
Cool. Thanks for the heads up, Greg!
